### PR TITLE
Update DeleteBehavior to allow client-only cascades

### PR DESCRIPTION
**Pull request by ajcvickers**
Issue #12661

### CODE REVIEW

File: src/EFCore/DeleteBehavior.cs [hunk 1]
```diff
@@ -63,36 +61,89 @@ public enum DeleteBehavior
...
+        ///     </para>
+        /// </summary>
+        NoAction,
+
+
+        /// <summary>
+        ///     <para>
+        ///         Note: it is unusual to use this value. Consider using <see cref="ClientSetNull"/> instead to match
+        ///         the behavior of EF6 with cascading deletes disabled.
+        ///     </para>
```
Comment (AndriySvyryd) [inline src/EFCore/DeleteBehavior.cs]: remarks section?

Comment (ajcvickers) [reply src/EFCore/DeleteBehavior.cs]: Last guidance we had was basically not to use the remarks section, but to instead use multiple paragraphs in the summary.

**Comment by ajcvickers** [comment]
Yep. Forgot to reference it in the commit message.

File: src/EFCore/DeleteBehavior.cs (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore/DeleteBehavior.cs]: We should make a pass to covert them throughout the code.

**Pull request merged**